### ISSUE TITLE
x-nullable / nullable / x-ms-nullable support with $ref in response schema

### ISSUE THREAD

**Issue by willemda** [Azure/autorest.typescript#1170]
AutoRest doesn't seem to support the following documented use-case (https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-nullable)

> ### Example: An operation that returns an object or null.
> 
> ```
> "responses": {
>   "200": {
>     "description": "The active Widget or null.",
>     "schema": {
>       "x-nullable": true,
>       "allOf": {
>         "$ref": "#/definitions/Widget"
>       }
>     }
>   }
> }
> ```

We've tried several variations of this (using `nullable`, `x-nullable` or `x-ms-nullable` next to `$ref` and/or without the `allOf`) with no luck.

Looking through existing autorest.typescript issues for nullable this looks like it should have been implemented following this issue: https://github.com/Azure/autorest.typescript/issues/680

**Comment by sarangan12** [comment]
I have analyzed the issue. The generator is handling the `x-nullable` when it is present within an object. But, when it is mentioned on the response, it does not have any effect. 

Now, @willemda I am wondering how you are expecting this to be handled? Because, the response could be null. Now, even without any further code modifications, it could be null. So, what do you actually expect in this case? Can you give an example?

@deyaaeldeen / @joheredi Do you remember any previous discussions relating to `x-nullable` on a response schema?

**Comment by willemda** [comment]
@sarangan12 We're love for this GET operation to explicitely define it can return a null response so that it reflects in the return type definition (`Widget | null`) 


The use-case is that for this GET operation a 200 or a 404 status code would share the same schema, but 404 would always be handled as `null`.

This would make it cleaner to handle non-existing objects when retrieving from an API (instead of handling an error we can check for `null` and it is clearly defined in the method contract)

**Comment by willemda** [comment]
@sarangan12 If I manually update the body mapper to set the `nullable` property to true, then this works as expected and my response doesn't get wrapped into a `body` property

```
export const WeatherForecast: coreClient.CompositeMapper = {
+++  nullable: true,
     type: {
     ....
}
```

It looks like this body mapper property is supported & used by the core-client library to determine if the response should be wrapped into a `body` property or not:

https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client/src/serviceClient.ts#L185
https://github.com/Azure/azure-sdk-for-js/blob/f89f1a97ce9658f2da50e31673fad805f7f42674/sdk/core/core-client/src/utils.ts#L119
https://github.com/Azure/azure-sdk-for-js/blob/f89f1a97ce9658f2da50e31673fad805f7f42674/sdk/core/core-client/src/utils.ts#L89

**Comment by joheredi** [comment]
@sarangan12 if we go with option 2, don't we also need to implement a change in the return type to tell that it can be `null` (option 1)?

**Comment by sarangan12** [comment]
Yes. Both the options need to be implemented to make the solution complete.
